packages/by-name/kata/kata-kernel-uvm/package.nix

```diff
@@ -20,28 +22,51 @@ let
       hash = "sha256-VcbOY86p8VkI6XvdhHfZNnWVHKuMLW7Xj7uzHHDiVsk=";
     };
 
-    postPatch = ''
-      config=$(find . -regex '.*/config-[0-9.-]+-confidential')
+    postPatch =
+      (
+        if withGPU then
+          ''
+            config=$(find . -regex '.*/config-[0-9.-]+-nvidia-gpu-confidential')
 
-      # 1. We don't use an initrd.
-      # 2. Enable dm-init, so that we can use `dm-mod.create`.
-      # 3. Disable module signing to make the build reproducable.
-      substituteInPlace $config \
-        --replace-fail 'CONFIG_INITRAMFS_SOURCE="initramfs.cpio.gz"' 'CONFIG_INITRAMFS_SOURCE=""' \
-        --replace-fail 'CONFIG_MODULE_SIG=y' 'CONFIG_MODULE_SIG=n' \
-        --replace-fail '# CONFIG_DM_INIT is not set' 'CONFIG_DM_INIT=y' \
-        --replace-fail '# CONFIG_DMIID is not set' 'CONFIG_DMIID=y' \
-        --replace-fail '# CONFIG_TMPFS_POSIX_ACL is not set' 'CONFIG_TMPFS_POSIX_ACL=y' \
-        --replace-fail '# CONFIG_TMPFS_XATTR is not set' 'CONFIG_TMPFS_XATTR=y' \
-        --replace-fail '# CONFIG_EFIVAR_FS is not set' 'CONFIG_EFIVAR_FS=y' \
-        --replace-fail '# CONFIG_RD_ZSTD is not set' 'CONFIG_RD_ZSTD=y' \
-        --replace-fail '# CONFIG_VFAT_FS is not se' 'CONFIG_VFAT_FS=y' \
-        --replace-fail '# CONFIG_NLS_CODEPAGE_437 is not set' 'CONFIG_NLS_CODEPAGE_437=y' \
-        --replace-fail '# CONFIG_NLS_ISO8859_1 is not set' 'CONFIG_NLS_ISO8859_1=y' \
-        --replace-fail '# CONFIG_ATA is not set' 'CONFIG_ATA=y'
+            # Enable dm-init, so that we can use `dm-mod.create`.
+            cat <<EOF >> $config
+            CONFIG_MD=y
```

Comment:
Is this needed, or is `CONFIG_BLK_DEV_DM` enough?

Reply:
These are from [upstream](https://github.com/kata-containers/kata-containers/blob/2c6126d3ab708e480b5aad1e7f7adbe22ffaa539/tools/packaging/kernel/configs/fragments/common/confidential_containers/cryptsetup.conf#L2). I'll add a comment indicating that.